webapp/channels/src/components/gif_picker/components/gif_picker_search.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ChangeEvent, memo, useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {useSelector} from 'react-redux';
+import tinycolor from 'tinycolor2';
+
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import {calculateContrastRatio} from 'utils/colors';
+
+import giphyWhiteImage from 'images/gif_picker/powered-by-giphy-white.png';
+import giphyBlackImage from 'images/gif_picker/powered-by-giphy-black.png';
+
+interface Props {
+    value: string;
+    onChange: (value: string) => void;
+}
+
+function GifPickerSearch(props: Props) {
+    const theme = useSelector(getTheme);
+
+    const [useDarkLogo, setDarkLogo] = useState(false);
+
+    const {formatMessage} = useIntl();
+
+    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+        event.preventDefault();
+
+        // remove trailing and leading colons
+        const value = event.target?.value?.trim()?.toLowerCase()?.replace(/^:|:$/g, '') ?? '';
+        props.onChange(value);
+    };
+
+    useEffect(() => {
+        const WHITE_RGB = [255, 255, 255];
+        const BLACK_RGB = [0, 0, 0];
+
+        const backgroundColorRGB = tinycolor(theme.centerChannelBg).toRgb();
+        const backgroundColor = [backgroundColorRGB.r, backgroundColorRGB.g, backgroundColorRGB.b];
+
+        const contrastRatioForBlack = calculateContrastRatio(backgroundColor, BLACK_RGB);
+        const contrastRatioForWhite = calculateContrastRatio(backgroundColor, WHITE_RGB);
+
+        if (contrastRatioForBlack > contrastRatioForWhite) {
+            setDarkLogo(true);
```

Comment:
Minor nit, but we don't need to store this in state. We could just calculate it in the outer function and wrap it in a `useMemo` if we think the computation is expensive enough to have an impact on performance. As-is, we might briefly show the black logo before switching to the white logo if the user is on a dark theme because the component would render twice.

Reply:
replaced with memo